src/com/amazon/corretto/crypto/provider/EcUtils.java

```diff
@@ -127,9 +146,22 @@ private static native int curveNameToInfo(String curveName,
             // Generator
             byte[] gx, byte[] gy,
             // Order of the generator
-            byte[] order);
+            byte[] order,
+            // String representation of OID; AWS-LC doensn't specify encoding but it appears to be ASCII/UTF-9
+            byte[] oid,
+            // DER-encoded info: OID for named, ECParameters for explicit (cf. RFC-3279 2.3.5)
+            byte[] encoded);
     private static native long buildGroup(int nid);
     private static native void freeGroup(long ptr);
+    private static native String[] getCurveNames();
+    private static native String getCurveNameFromEncoded(byte[] encoded);
+
+    static String getOidFromName(String name) {
+        if (name == null) {
+            return null;
+        }
+        return getSpecByName(name) == null ? null : getSpecByName(name).oid;
```

Comment:
Maybe first store `getSpecByName` to avoid invoking it again when it's not null.

Reply:
yeah might as well save a method invocation. will fix.